### ISSUE TITLE
Android Hybrid (Visualforce) app crashes upon startup due to timeout in fontdoor.jsp login dance

### ISSUE THREAD

**Issue by daveespo** [forcedotcom/SalesforceMobileSDK-Android#2386]
Please fill out the following details:

1. Version of Mobile SDK Used: 10.0.2
2. Issue found in Native App or Hybrid App: Hybrid
3. OS Version: Android 11
4. Device: Samsung Tab A
5. Steps to reproduce: Launch the app and complete the Salesforce login -- accept the Connected App authorization screen
6. Actual behavior: Popup modal that said:
```
Application Error - The connection to the server was unsuccessful

https://ptqa1925.my.salesforce.com/secur/frontdoor.jsp?&sid=***
```

 Expected Behavior: Expected to log in and land at the Visualforce page defined in bootconfig
10. Error Log: `CordovaWebView: TIMEOUT ERROR!`

I redacted the URL above due to it containing a sensitive session ID.

I was able to take the URL that is identified in the error message above (the whole thing with the Session ID and retURL) and load it in a regular web browser on a PC and saw that the redirect through `contentDoor` was taking 20 seconds:

![image](https://user-images.githubusercontent.com/233309/221214839-e51f9654-7604-438a-b106-4a7db2b77cd2.png)

The default timeout for CordovaWebViewImpl is 20 seconds so the 20+ second wait on the `contentDoor` was causing the timeout during app launch. 

**Note** This only happens in some orgs. I could not repro in DE orgs. I could only repro in EE orgs. All orgs have Enhanced Domains enabled so I was unable to find an obvious explanation in Setup to explain why I saw different timings for the login dance. 

Workaround:

You can define a preference in config.xml to increase the timeout:

```
<preference name="LoadUrlTimeoutValue" value="60000" />
```

Given the erratic timing of the login dance, can you increase the default timeout in `CordovaWebViewImpl line 148` to 60 seconds?

**Comment by gkotula** [comment]
Hi @daveespo, globally increasing the timeout would need some consideration before we decide to go that route.  Is this issue consistently reproducible for certain Orgs across multiple networks (solid WiFi, cellular, etc.)?  If the Org is problematic, there may be a solution with configuration for those Orgs.

**Comment by daveespo** [comment]
@gkotula -- I had multiple end users report the same behavior across different orgs so it can't be correlated to local connectivity. Additionally, the screenshot I pasted above was from a wired gigabit connection from a conventional web browser (Chrome) -- so this is definitely on the Salesforce server side.

At the same time where EE orgs were exhibiting this problem, the DE orgs that I was using for development were NOT exhibiting the same problem. All orgs had Enhanced Domains enabled and deployed. I spent a fair amount of time trying to reason about what the configuration difference could be between the DE and EE orgs but couldn't come up with anything. 

I appreciate that you can't modify `CordovaWebViewImpl` but you could add the `LoadUrlTimeoutValue` to the boilerplate `config.xml` that is generated when generating a new app template. Also, I have to say that for Salesforce and Visualforce, in particular, 20 seconds is unfortunately a rather short timeout. A "cold" Visualforce page can take up to 30 seconds to compile and render server-side. 

Like I said, I have my workaround in place and I'm hoping that others that encounter the same thing provide support for getting the timeout increased to avoid future frustration of app crashes in the field.

**Comment by daveespo** [comment]
Correct, I have a reasonable workaround at this time.

**Comment by gkotula** [comment]
Good to hear.  For now, I've added the enhancement label, and we have this proposal tracked in our internal ticketing system.  Thank you for the suggestion!

**Comment by gkotula** [comment]
@daveespo I'm connecting some dots here, and you are not the only one who has recently had issues with the performance of that endpoint... Could you verify whether all the problematic Orgs have been upgraded to Spring '23 release?

On a quick note, it does seem valuable to give that config option some more exposure.  I'm looking into whether that would be a good quick win for the SDK, but we have to be careful about tweaking default values which may impact all of our consumers.

**Comment by daveespo** [comment]
All orgs have been upgraded to Spring '23 so that wouldn't explain the difference between the orgs I've tested with. 

Also to note, an interactive login to these orgs (username/password in a desktop browser) is speedy and there is no delay in the login dance greater than 1-2 seconds. It seems to be unique to the login dance using the session ID.

**Comment by gkotula** [comment]
@daveespo Could you share what products the problematic Orgs are subscribed to?  Marketing, Commerce, etc.?  Knowing how your Orgs are configured may help expedite the solution.

**Comment by daveespo** [comment]
They are just plain vanilla Trial EEs that have Sales cloud only.

**Comment by wmathurin** [comment]
Closing this issue as it is a duplicate of https://github.com/forcedotcom/SalesforceMobileSDK-Android/issues/2386.
We are looking into moving away from frontdoor when loading the remote page in an hybrid remote app. We will update on the other issue.

**Comment by daveespo** [comment]
@wmathurin  -- you closed this issue by referencing the same issue as a duplicate :-) What issue did you intend to reference with your prior comment?

**Comment by wmathurin** [comment]
Good catch - I updated the earlier comment with the correct link.
